code-challenges.js

```diff
@@ -30,35 +44,65 @@ var randomNouns = ["streetlamp", "potato", "teeth", "conclusion", "nephew", "tem
 var mixedDataArray = [true, 8, "hello", 90, -8, null, 0, 46, 59, 107, "hey!"]
 // Expected output: [-8, 0, 8, 46, 59, 90, 107]
 
+const onlyNumbers = (array) => {
+    let newArr = []
+    newArr = array.sort((a,b) => a - b).filter(value => typeof value === "number")
+     return newArr 
 
-
+}
+console.log(onlyNumbers(mixedDataArray))
 
 
 // --------------------4) Create a function that takes in a string and logs the index of the first vowel.
 // Use the test variables provided.
 
 var vowelTester1 = "learn"
 // Expected output: 1
-var vowelTester1 = "throw"
+var vowelTester2 = "throw"
 // Expected output: 3
 
+// create a function called vowelIndex that takes in a string and logs the index of the first vowel,
+// create an array called vowelArray that contains only vowel letters.
 
+const vowelIndex = (str) => {
+   let vowelArray = ["a", "e", "i", "o", "u"]
+    str.split("")
+        for (let i= 0; i <str.length; i++)
+            if (vowelArray.includes(str[i])) {
+                return i
+            }
```

Comment:
so i think you're missing one curly brace here. with some small changes to your indentation you'll be able to see where :) also, you don't really need to split the string. strings can be iterated through with loops just like arrays. cool huh?